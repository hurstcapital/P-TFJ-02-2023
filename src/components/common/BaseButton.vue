<template>
  <button
    data-cy="submit"
    type="submit"
    :disabled="isDisabled"
    class="bg-green-500 hover:bg-green-700 text-white font-bold py-2 px-3 rounded"
    :class="
      isDisabled
        ? 'focus:outline-none disabled:opacity-50 cursor-not-allowed hover:bg-green-500'
        : ''
    "
  >
    <slot></slot>
  </button>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "BaseButton",
  props: {
    isDisabled: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
});
</script>
