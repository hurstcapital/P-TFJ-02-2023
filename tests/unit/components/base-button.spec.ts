import { shallowMount } from "@vue/test-utils";
import BaseButton from "@/components/common/BaseButton.vue";

describe("BaseButton.vue", () => {
  it("should render slot", () => {
    const slot = "slot example";

    const wrapper = shallowMount(BaseButton, {
      slots: {
        default: slot,
      },
    });

    const textSlot = wrapper.find("button").text();

    expect(textSlot).toBe(slot);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("shloud be disabled", () => {
    const wrapper = shallowMount(BaseButton, {
      props: {
        isDisabled: true,
      },
    });

    expect(wrapper.find("button").attributes("disabled")).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  });
});
