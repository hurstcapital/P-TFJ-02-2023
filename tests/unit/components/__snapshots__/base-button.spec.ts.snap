// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton.vue shloud be disabled 1`] = `
<button
  class="bg-green-500 hover:bg-green-700 text-white font-bold py-2 px-3 rounded focus:outline-none disabled:opacity-50 cursor-not-allowed hover:bg-green-500"
  data-cy="submit"
  disabled=""
  type="submit"
>
  
  
</button>
`;

exports[`BaseButton.vue should render slot 1`] = `
<button
  class="bg-green-500 hover:bg-green-700 text-white font-bold py-2 px-3 rounded"
  data-cy="submit"
  type="submit"
>
  
  slot example
  
</button>
`;
